concent_api/concent_api/system_check.py

```diff
@@ -613,3 +677,47 @@ def check_concent_ethereum_public_key(app_configs: None=None, **kwargs: Any) ->
             return [create_error_47_concent_ethereum_public_key_has_wrong_length(settings.CONCENT_ETHEREUM_PUBLIC_KEY)]
 
     return []
+
+
+@register()
+def check_middleman_address(app_configs: None=None, **kwargs: Any) -> list:  # pylint: disable=unused-argument
+    if not hasattr(settings, 'MIDDLEMAN_ADDRESS') and 'middleman' in settings.CONCENT_FEATURES:
+        return [create_error_49_middleman_address_is_not_set()]
+
+    if hasattr(settings, 'MIDDLEMAN_ADDRESS'):
```

Comment:
I think I have discussed it recently with @cameel and we did this only for few settings for verification. Generally that should not cause any problems, and there is a lot of other settings which work this way. But if you insist I can change it - please just propose the error message.
@cameel Can you confirm if we should be strict about it?

Reply:
The thing we talked about was whether to require the setting to be undefined (or set to an empty value in some cases) when the feature that uses it is not enabled. We have many containers running with slighly different `local_settings.py` in deployment and this would allow us to spot a mistake such as putting the value in the wrong configuration file. Initially my opinion was that this should be enforced and I was including such a check when defining issues. But then we have discovered a case where it's really inconvenient - when you want to have a global default in `base.py`. Getting an error simply because you did not remove a setting from `base.py` with `del` in your `local_settings.py` seems like a very unfriendly and possibly confusing thing to do. You could get that error even on a clean config with no extra settings added.

So eventually I stopped recommending it. A setting from a feature can be empty when the feature is disabled but does not have to.

The case here is slightly different. You're asking if the setting should be validated if you do decide to define it. My opinion is that it should. There's no reason to have an invalid value in it. If you can't provide a valid value when the feature is disabled then just don't set it.